gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/proc/EnforceJobStartDeadlineDagProc.java

```diff
@@ -0,0 +1,102 @@
+import org.apache.gobblin.service.modules.spec.JobExecutionPlan;
+
+import static org.apache.gobblin.service.ExecutionStatus.ORCHESTRATED;
+import static org.apache.gobblin.service.ExecutionStatus.valueOf;
+
+
+/**
+ * An implementation for {@link DagProc} that marks the {@link Dag} as failed and cancel the job if it does not start in
+ * {@link org.apache.gobblin.service.modules.orchestration.DagManager#JOB_START_SLA_TIME} time.
+ */
+@Slf4j
+public class EnforceJobStartDeadlineDagProc extends DagProc<Optional<Dag<JobExecutionPlan>>> {
+
+  public EnforceJobStartDeadlineDagProc(EnforceJobStartDeadlineDagTask enforceJobStartDeadlineDagTask) {
+    super(enforceJobStartDeadlineDagTask);
+  }
+
+  @Override
+  protected Optional<Dag<JobExecutionPlan>> initialize(DagManagementStateStore dagManagementStateStore)
+      throws IOException {
+   return dagManagementStateStore.getDag(getDagId());
+  }
+
+  @Override
+  protected void act(DagManagementStateStore dagManagementStateStore, Optional<Dag<JobExecutionPlan>> dag)
+      throws IOException {
+    log.info("Request to enforce deadlines for dag {}", getDagId());
+
+    if (!dag.isPresent()) {
+      // todo - add a metric here
+      log.error("Did not find Dag with id {}, it might be already cancelled/finished and thus cleaned up from the store.",
+          getDagId());
+      return;
+    }
+
+    enforceJobStartDeadline(dagManagementStateStore, dag);
+  }
+
+  private void enforceJobStartDeadline(DagManagementStateStore dagManagementStateStore, Optional<Dag<JobExecutionPlan>> dag)
+      throws IOException {
+    Pair<Optional<Dag.DagNode<JobExecutionPlan>>, Optional<org.apache.gobblin.service.monitoring.JobStatus>>
+        dagNodeToCheckDeadline = dagManagementStateStore.getDagNodeWithJobStatus(getDagNodeId());
+    if (!dagNodeToCheckDeadline.getLeft().isPresent()) {
+      // this should never happen; a job for which DEADLINE_ENFORCEMENT dag action is created must have a dag node in store
+      log.error("Dag node {} not found for EnforceJobStartDeadlineDagProc", getDagNodeId());
+      return;
+    }
+
+    Dag.DagNode<JobExecutionPlan> dagNode = dagNodeToCheckDeadline.getLeft().get();
+    long timeOutForJobStart = DagManagerUtils.getJobStartSla(dagNode, DagProcessingEngine.getDefaultJobStartSlaTimeMillis());
+    Optional<org.apache.gobblin.service.monitoring.JobStatus> jobStatus = dagNodeToCheckDeadline.getRight();
+    if (!jobStatus.isPresent()) {
+      return;
+    }
+
+    ExecutionStatus executionStatus = valueOf(jobStatus.get().getEventName());
+    long jobOrchestratedTime = jobStatus.get().getOrchestratedTime();
+    // note that second condition should be true because the triggered dag action has waited enough before reaching here
+    if (executionStatus == ORCHESTRATED && System.currentTimeMillis() > jobOrchestratedTime + timeOutForJobStart) {
+      log.info("Job {} of flow {} exceeded the job start SLA of {} ms. Killing the job now...",
```

Comment:
I suggest a similar change in log message to put a cohesive message beforehand that's easily searchable then all the parameters at the end